capability.go

```diff
@@ -214,18 +224,19 @@ func (c Client) startCall() (hook ClientHook, resolved, released bool, finish fu
 	if c.h == nil {
 		return nil, true, c.released, func() {}
 	}
-	c.h.mu.Lock()
-	c.h = resolveHook(c.h)
+	l := c.h.state.Lock()
+	c.h, l = resolveHook(c.h, l)
 	if c.h == nil {
 		return nil, true, false, func() {}
 	}
-	c.h.calls++
-	c.h.mu.Unlock()
+	l.Value().calls++
+	isResolved := l.Value().isResolved()
```

Comment:
This here is the only functional change; previously isResolved was called in-line in the return statement, but this is incorrect according to isResolved()'s docs, which say you must be holding the lock. Making this change turned that into a type error, which brought it to my attention.